### DISCUSSION TITLE
Anbindung JK BMS

### DISCUSSION THREAD

**Discussion by iukbox** [hoylabs/OpenDTU-OnBattery#415] [Q&A]
Ich habe gerade mit Freude gesehen das ihr das JK BMS supported.

Gibt's einen Anschlussplan und eine ( falls erforderlich ) pinmaping Datei oder Beispiel dazu?
Dann hätte ich endlich alles in der HA.

**Comment by schlimmchen** [comment]
Das ist ja erfreulich, dass sich ein weiterer potentieller User meines Features meldet :blush:

* Schau mal ins [Wiki](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Jikong-JK-BMS). Reicht dir das als Anschlussplan, ist das verständlich? Im Wiki findest du auch die nötigen Infos zwecks Pinmapping.
* Das Feature ist inzwischen Teil des development Branches dieses Projekts, aber noch nicht in einem Release enthalten. Baue dir entweder selbst eine Firmware, warte auf ein Release, oder schau dir [mein Vorab-Release](https://github.com/schlimmchen/OpenDTU-OnBattery/releases/tag/2023.08.29-jkbms) an.
* Integration in Home Assistent ist noch nicht vollständig. Ich hab mit Home Assistent nichts am Hut bisher. Ich lasse mich da mal bootstrappen von ChatGPT und prüfe, ob ich da nicht sogar etwas kaputt gemacht habe...
* Meinst du mit "HA" nicht Home Assistent sondern Home Automation? Bisher wird nur der Hersteller, das Alter der Daten und die State of Charge über MQTT übermittelt.

**Reply by iukbox** [reply]
Mit HA meinte ich Hausautomation.
Die Infos im Wiki sind zu viel Input auf einmal für mich. Da blicke ich im Moment nicht durch. Ist das jetzt einfach ne serielle Kommunikation oder was anderes?

**Reply by schlimmchen** [reply]
Naja, ich hab ja schon den richtigen konkreten Wiki-Artikel verlinkt, den schaffst du schon zu verdauen. Am Ende ist es eine serielle Schnittstelle, korrekt.

**Comment by iukbox** [comment]
Gibt's eine Planung wann du die kompletten Daten per MQTT überträgst?
Die Variante mit dem JK RS485 Adapter an der Fusion Platine ist aber getestet und funktioniert? Wenn ich das jetzt angehe, würde ich das auch direkt auf verlässliche Hardware setzen.

**Comment by schlimmchen** [comment]
> Die Variante mit dem JK RS485 Adapter an der Fusion Platine ist aber getestet und funktioniert? Wenn ich das jetzt angehe, würde ich das auch direkt auf verlässliche Hardware setzen.

Ja, das funktioniert sehr gut. Die OpenDTU-Fusion Platine ist toll :wink:

> Gibt's eine Planung wann du die kompletten Daten per MQTT überträgst?

Ne. Aber ich hab eine Idee zur Umsetzung. Ist sehr viel los gerade bei mir.

**Comment by schlimmchen** [comment]
@iukbox Siehe #432. Kannst du das testen (meinen Branch auszuchecken und ihn zu bauen und bei dir einzusetzen)? Hast du inzwischen eine OpenDTU-FUSION Platine und die Anbindung and JK BMS stehen?

**Reply by iukbox** [reply]
Platine hatte ich noch nicht bestellt weil ich deine Antwort abwarten wollte.
Ich blicke durch die GitHub Funktionen und Begriffe nicht durch ( Branch / auschecken etc. ) , hab bisher nichts damit zu tun gehabt außer *.bin Dateien Download und diese Diskussionen hier. Die überwiegend englische Sprache tut ihr übriges. 😄

**Reply by schlimmchen** [reply]
Verstanden, kein Problem. Das Veröffentlichen über MQTT funktioniert gut und ich werde #432 für reif erklären. Halte ein Auge auf die nächsten Releases und schau, ob du im Changelog dieses Feature siehst, dann ist es auch enthalten.

**Reply by Andy0043** [reply]
Hallo, ich habe 2 Jk BMS im Einsatz. Steuere im Haus vieles über Node Red und mit Tasmota geflashten Bauteilen. Ich wäre sofort dabei dieses Projekt umzusetzen. Könnte dann Steuern und messen. Daten in die InfluxDB schreiben und noch mit Grafana visualisieren. Lese schon meine Zählerdaten aus und visualisiere diese.
Und dann muss ich nur noch an die Daten meiner Wechselrichter kommen. Ich habe 3 Studer XTM 4000-48. Da geht ja irgendwann  etwas über RS232 oder den  CAN Bus. Aber das wäre dann ein anderes Thema.
Bitte unbedingt weiter machen.
Gruß Andreas

**Reply by iukbox** [reply]
Sodele, PLatine und original JK BMS Adapter sind angekommen.. Jetzt muss ich mal suchen welche Farbe vom RS485 Adapter wo an die Platine kommt, oder hat das schon jemand raus gefunden?

**Reply by schlimmchen** [reply]
Liest denn keiner das [Wiki](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Jikong-JK-BMS)? 😉 Y, W, und B stehen für yellow, white, black. Sollte das deutlicher beschrieben sein? Kann das gerne noch erläutern, dass das für die Farben steht. Es geht um die ASCII-art.

**Reply by iukbox** [reply]
Sorry, kann noch nichts zur Funktion sagen, das OpenDTU Fusion Board will nicht so wie ich. Es läßt sich flashen mit den Einstellungen in den Docs, jedoch öffnet es nach der Installation keinen AP. Auch nach mehreren Versuchen blieb es erfolglos.
Ich komme da im Moment nicht weiter.

**Reply by schlimmchen** [reply]
Schade. Brauchst du Hilfe? Du hast die Firmware für den ESP32-S3 genommen und zwar die factory-Variante? Wie bekommt das Board Strom nach dem Aufspielen der Firmware? WiFi-Antenne hast du angeschlossen?

**Reply by iukbox** [reply]
Firmware für den ESP32-S3 in der factory usb Variante, WiFi Antenne ist dran, Strom bekommt der ESP noch über de USB Buchse, Jumper steht auch noch auf USB.

**Reply by schlimmchen** [reply]
Hm, das ist ja Mist. Dann hab ich leider keinen besseren Rat als zusätzlich einmal den Flash des ESP32 zu löschen, um sicherzustellen, dass da kein Kram drinsteht, der da nicht hingehört. Hast du das Board gebraucht gekauft? Was siehst du denn auf der seriellen Konsole, wenn das Ding startet?

**Reply by iukbox** [reply]
Mittlerweile hats geklappt, komischerweise wurde der AP an dem Rechner der sonst immer für solche Aufgaben genutzt wurde NICHT angezeigt.
An einem Laptop wurde er dann angezeigt, alles dann eingerichtet, neueste Software und Pinmapping drauf und dann die Tage mal ans BMS hängen. Bin leider derzeit am renovieren und hab nicht so viel Zeit übrig wie ich möchte.

**Reply by iukbox** [reply]
Hab heute mal Zeit bekommen das Fusion Board mit dem originalen JK BMS Adapter anzuschließen laut Plan.
Komisch 😃
Funktioniert direkt 😄
Daten werden im Web Frontend und per MQTT übertragen.

**Reply by iukbox** [reply]
Danke an @schlimmchen für die Arbeit.
Jetzt muss ich nur noch prüfen ob der errechnete SOC vom BMS stimmt damit ich den Akku mal voll ausnutzen kann.

**Reply by schlimmchen** [reply]
@iukbox Schau dir mal https://github.com/helgeerbe/OpenDTU-OnBattery/issues/507#issuecomment-1789497670 an. Also der SoC passt nur dann zu deiner Batterie, wenn die Einstellungen (Schwellenwerte) des BMS wirklich zur Batterie passen. Sonst nimmt das BMS an, dass beim Erreichen der Undervoltage-Protection-Threshold die Batterie auf 0% SoC ist, obwohl dem tatsächlich keineswegs so ist.

**Reply by iukbox** [reply]
Wird eigentlich irgendwas vom JK BMS auf dem Display angezeigt wenn das in der openDTU aktiviert ist?
Konnte das bisher noch nicht testen da die Hardware im Realeinsatz ist.

**Reply by iukbox** [reply]
Ich kann jetzt nicht erkennen, welche Werte für den Akku im BMS da richtig sein sollen damit der SOC korrekt berechnet wird.

Derzeit habe ich bei 3,31V ( im Durchschnitt )  einen SOC von 82%, ich kann's ja Dank deiner Lösung visualisieren :)
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/26631519/60031520-e2aa-4128-bbd5-b346fc5ebd34)

**Reply by schlimmchen** [reply]
> Wird eigentlich irgendwas vom JK BMS auf dem Display angezeigt wenn das in der openDTU aktiviert ist?

Das Display an der OpenDTU? Nein, da gibt es keine Infos von der Batterie.

> Ich kann jetzt nicht erkennen, welche Werte für den Akku im BMS da richtig sein sollen damit der SOC korrekt berechnet wird.

Wenn du das BMS einmal initialisiert hast und den "LiFePo4" Button gedrückt hast und danach die Schwellenwerte nicht unsinnig (wie ich) angepasst hast, sollte es passen.

**Reply by iukbox** [reply]
Ich hab gestern durch Zufall beobachten können wie ( beim Ladevorgang ) plötzlich der SOC von 90 auf 100% sprang ob wohl die Voltzahl mit ca. 53 Volt nicht passte ( Schwellwerte sind alle noch original ). Ich fahre die Einspeisung mal runter und lasse den mal bis zur Absorption mit den Victron hoch laufen.

**Comment by iukbox** [comment]
@schlimmchen 
Ist es geplant die einzelnen Zellspannungen und das Log per MQTT zu übertragen?
Ich weiß, es ist ne Menge Traffic aber man könnte das ja auch einen anderen Rhythmus beschränken. Für die Prüfung des Akku wäre das prima, dann brauchte man nicht immer in die Nähe des BMS zu laufen.

**Reply by schlimmchen** [reply]
![cell_voltages](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/3578416/ca35952a-83df-40f3-a85b-dfee557923f3)

Das läuft bisher nur auf meinem staging-OpenDTU-FUSION board und es sind Dummy-Daten. Aber im Grunde ist (unter anderem) genau das, wonach du fragst, schon fertig. Ich schau mal, dass ich das demnächst in einem pull request verpacken kann.

**Reply by schlimmchen** [reply]
@iukbox Schau mal hier: https://github.com/helgeerbe/OpenDTU-OnBattery/actions/runs/7214520701?pr=549 Diese Artefakte beziehen sich auf pull request https://github.com/helgeerbe/OpenDTU-OnBattery/pull/549 und der sollte deinen Wunsch erfüllen :blush: Gib gerne Feedback.

**Reply by iukbox** [reply]
![Screenshot_2023-12-20-13-57-19-611_xyz wallpanel app](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/26631519/05977771-64db-493e-b464-074d02ca5381)
Schon eingebunden, jetzt hab ich es wenigstens im Blick.
Danke für deine / eure Arbeit.

**Comment by iukbox** [comment]
Danke funktioniert top.
Hab gerade ein neues BMS montiert ( die 150 a Variante ) , die funktioniert auch einwandfrei. Es scheint als hätten sie an der Software was geändert denn direkt nach dem Abschluss sieht es so aus als würde der SoC stimmen.
Das werde ich mal im Auge behalten

**Comment by iukbox** [comment]
@schlimmchen 
Sind die Daten eigentlich nur zum lesen oder kann ich auch Daten schreiben, z.b. Discharge ausschalten?

**Reply by schlimmchen** [reply]
Bisher wird nur gelesen. Der Code im Hintergrund ist aber dafür gemacht, dass man relativ leicht auch Nachrichten/Befehle an das BMS schicken kann. Trotzdem fehlt da noch einiges, bis das über die WebUI bedient werden kann... Es hängt von mir aus nur an Zeitmangel...

**Reply by iukbox** [reply]
Ja, GUI ist immer die meiste Arbeit.
MQTT würde ja auch als erster Schritt mal reichen 😀

**Comment by gitisgreat2023** [comment]
Which JK BMS are supported?

For example JK Smart Active Balance BMS BD6A17S6P - (8S - 17S) ?

**Reply by greymda** [reply]
@gitisgreat2023 have you had a chance to figure it out eventually? cheers

**Comment by schlimmchen** [comment]
There are no guarantees. However, I suspect that all JK BMS use the same protocol.

These are the names explicitly listed in the manual that I have, all of which are advertised to have an RS485 interface:

JK-BD6AxxS-6P/ JK-BD6AxxS-8P
JK-BD6AxxS-10P / JK-BD6AxxS-15P
JK-BD6AxxS-20P/ JK-B1AxxS-15P
JK-B2AxxS-15P/ JK-B2AxxS-20P

Notice that the RS485 feature conflicts with the CAN feature. If you bought a version with a CAN interface, the BMS will have **no** RS485 interface.

Have a look at [the Wiki](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Jikong-JK-BMS) which I will update to list the supported devices.

**Reply by eeprom23** [reply]
![IMG_3460](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/37772803/d3a1ab70-1f50-4c5e-9cb5-8acd5594dfad)


Muss ich den rs485 oder den GPS Port nutzen ?
Das bms ist ein b2a8s20p

oder geht es nicht mit der opendtu ?

**Reply by schlimmchen** [reply]
Schau mal im Wiki: https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Jikong-JK-BMS#rs485-ttl-uart-jack Da habe ich ein Foto ergänzt. Du möchtest an den als "GPS" beschrifteten Port. Kannst du mit einem Multimeter herausfinden, welcher der Pins GND und welcher B+ ist? Im Wiki (Link oben) habe ich eine Notiz ergänzt, dass das Pin-Out bei diesen "silbernen" BMS andersherum zu lesen ist (bzw. muss man auf die angedeuteten Schlitze achten). Ich würde gerne wissen, ob das korrekt ist, also ob bei deinem BMS tatsächlich der "linke" (wie in deinem Foto) Pin B+ ist und der "rechte" GND. Danke!

**Reply by eeprom23** [reply]
Es ist so wie du es geschrieben hast, der linke ist plus (b+) und der rechte minus ! 

![IMG_3465](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/37772803/59e28cd4-2164-4623-aa5b-5e429ba94060)

**Reply by schlimmchen** [reply]
@eeprom23 Danke für die Rückmeldung! Ich hab dein Bild geklaut fürs Wiki. Melde dich, falls das nicht okay ist.

**Reply by eeprom23** [reply]
Es wäre noch hilfreich im Wiki zu erwähnen , dass es JST 1,25mm 4 polige Stecker sind welche für den GPS Port passend sind. Anbei noch ein paar links von Amazon welche passen.

https://amzn.eu/d/3Dw0yR6

https://amzn.eu/d/0ewtjrF

https://amzn.eu/d/hfY1rc1

**Reply by schlimmchen** [reply]
Steht doch da:

![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/3578416/7b48958a-7a9a-4495-bc91-dd162a57ca02)

**Comment by Tu4r3g21** [comment]
Hi,
Ich wollte mich auch für die phantastische Arbeit bedanken! Seit zwei Wochen werkelt der Akku in der Gartenhütte mit dem JK BMS und da aktuell nur ein Teil der PV Module angeschlossen sind und auch nur nach Westen schauen, während sie im Schatten stehen, wurde heute zum ersten Mal die obere Ladeschlussspannung von 3.45V erreicht (nur kurz).
Andy von OpenGridGarage hat kürzlich ein interessantes Video online gestellt, in dem er erklärt wie man prinzipiell jedes JK BMS auf 100% SoC kalibrieren kann (z.B. OVP auf 3.451V, warten bis voll und dann wieder auf 3.65V) falls dieser Wert bei dem ein oder anderen ab und zu schwankt. Ich denke die werden sich alle auf 100% resetten, sobald min. eine Zelle auf OVP geht, aber soweit will man ja eigentlich nicht.
Ich fände es auch Klasse, wenn man Laden-/Entladen. Notfall etc. Über die DTU steuern (aktivieren/deaktivieren) könnte. Da ich aber selber keinen Beitrag dazu leisten kann, weil mir die Skills fehlen, übe ich mich in Geduld und hoffe! 😉👍
Also nochmal, vielen Dank die tolle Umsetzung und die überaus umfängliche Wissenssammlung, die es sogar einem Maschinenbauer wie mir ermöglicht sowas umzusetzen 😂.

**Reply by schlimmchen** [reply]
> Ich fände es auch Klasse, wenn man Laden-/Entladen. Notfall etc. Über die DTU steuern (aktivieren/deaktivieren) könnte.

Fände ich auch interessant, aber nur als Spielerei. In welchen Situationen würdest du das nutzen?

> Andy von OpenGridGarage hat kürzlich ein interessantes Video online gestellt, in dem er erklärt wie man prinzipiell jedes JK BMS auf 100% SoC kalibrieren kann

Off-Grid Garage? Das hier: https://www.youtube.com/watch?v=maTqnSs_XAI ?

Hast du schon gelesen, das mein neues JK BMS mit neuerer Firmware neue Parameter "100% SoC" und "0% SoC" hat? Ich wünschte echt, die "alten" JK BMS könnte man updaten. Leider scheint das nicht zu gehen.

Ich sehe, dass Andy das auch beschreibt und ein neueres JK BMS hat mit diesen Einstellungen. Bei ihm funktionieren die aber nicht. Allerdings hat er auch Firmware 11.281, meins hat 11.288, vielleicht funktioniert es bei meinem (noch nicht ausprobiert). Der Reset auf 0% bei der von mir eingestellten Spannung scheint jedenfalls zu funktionieren. In den Kommentaren zum Video liest man, dass zwei andere auch 11.288 haben und es da funktioniert mit dem Reset auf 100%. Das BMS für meinen Vater ist erst vor zwei Tagen bestellt und ich hoffe, dass es auch eine junge Firmware hat. Trotzdem bin ich zurück beim Steuern des DPL über Spannungen und hab dafür (nach einer Diskussion mit @hoschiking) die Funktion eingeführt, dass man trotz angeschlossenem BMS den DPL per Spannung statt SoC steuern kann, sodass man vermeidet "ständig" in OVP oder UVP zu rennen.

Andy schwärmt ja auch von den JK Inverter BMS, die auch Firmware Updates erhalten können... Die scheinen interessant zu sein.

**Reply by hoschiking** [reply]
Das Video kannte ich noch nicht, aber im Grunde macht er es nicht anders als von mir vorgeschlagen, bis auf die sinnlose Umstellung auf 3,65V. Es empfiehlt sich es auf 3,45V zu belassen und so den SOC regelmässig zu kalibrieren. Ja, es ist eine Krücke, aber anders geht es nicht, es sei denn, man hat eines mit neuer Firmware erwischt, bei dem man die 2 neuen zusätzlichen Werte hat. JK hat es ebenfalls selbst gemerkt, dass es vorher nur suboptimal war.

**Comment by Tu4r3g21** [comment]
😂 frag mich nicht was meine Finger da getippt haben... Aber klar, OffGridGarage war gemeint... 🫣
Und auch genau das Video. Ich muss zugeben, ich hab mich von ihm einfach ein wenig anstecken lassen bzgl. der JK BMSe (BMSns?) Da offensichtlich 100% mehr Erfahrung mit diversen Herstellern als ich. Und ich muss sagen, für das erste Mal das ich einen Akku gebaut habe, lief  mit dem JK alles absolut problemlos und genau wie erwartet.

Ich hatte heute die DTU und meine Grafanaseite ständig im Auge um die Einzelspannungen zu beobachten, während das Pack zum ersten Mal Richtung 55.2V unterwegs war. Dabei kam mir der Gedanke, das ich von draussen keine Möglichkeit hätte, den Akku abzuschalten, wenn ich physisch den Trennschalter nicht bedienen kann weil unterwegs oder so. Es gibt bestimmt einige Situationen an die ich bisher noch gar nicht denken kann, an denen man sich einen Remoteswitch wünschen würde oder?

**Comment by hoschiking** [comment]
Es ist durchaus empfehlenswert, einen Remote-PIN zu implementieren. Ich vermisse es auch. Sowas sollte in der Regel immer vorhanden sein, der lässt sich vielfältig einsetzen. Da gibt es genug Usecases dafür. An jedem guten Netzteil, Ladegerät und ähnlichem gibt es einen ;-)

**Reply by schlimmchen** [reply]
Schreib das mal bitte in ein neues Issue (feature request), dann wird es zumindest nicht vergessen.

**Comment by Tu4r3g21** [comment]
Ich habe nochmal eine Frage bzgl JK BMS. In der Live Ansicht werden ja auch die Namen der einzelnen Komponenten wie Victron oder Hoymiles angezeigt,  beim JK BMS steht aber dann bei mir "Batterie: Input Userda " und der Name den ich dem BMS per App vergeben habe. In der App sehe ich natürlich nur den Namen ohne den Zusatz. Gibt es eine Chance den Namen in der DTU noch irgendwo zu ändern, so das auch z.b. der mqtt Name auch etwas mehr "user-friendly" ist?

**Reply by schlimmchen** [reply]
Merkwürdig. Teile mal bitte einen Screenshot.

**Reply by Tu4r3g21** [reply]
![Screenshot_20240413-153628](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/4610513/8aca7560-034a-4160-9510-6320d75b085c)

**Reply by Tu4r3g21** [reply]
KWB-BAT-01 ist der Name den ich in der App per Bluetooth dem BMS gegeben hab.

**Reply by schlimmchen** [reply]
Ja, verstehe, ich kann das reproduzieren. Weiter in #890.

**Comment by Tu4r3g21** [comment]
Hi, Ich wieder....
Ich teste gerade den Home Assistant und habe die Daten von der DTU, sowie vom Victron, dem Hoymiles und dem JK BMS verfügbar. Im iObroker hatte ich auch alle Einzelzellen und fand die Möglichkeit eigentlich sehr nett, eine einzelne 'driftende' Zelle schnell optisch in einem Diagramm identifizieren zu können. Im Home Assistant bekomme ich aber nur die maximale Spannungsdifferenz angezeigt. Ist das eine Frage der Settings (im MQTT Broadcast oder empfangene Topics?) oder liegt das evlt. an der HA Autodiscover Funktion? Ich nutze den Mosquitto MQTT. Ich kann mir eigentlich nicht vorstellen das es an der DTU liegen soll, denn an der habe ich nichts geändert, außer den Broker anzupassen (IP etc.).
Besten Dank schon mal im Voraus für Rückmeldungen!

**Reply by schlimmchen** [reply]
> Im iObroker hatte ich auch alle Einzelzellen und fand die Möglichkeit eigentlich sehr nett, eine einzelne 'driftende' Zelle schnell optisch in einem Diagramm identifizieren zu können.

Das klingt mir schwer nach einem ioBroker feature. Jedenfalls tut die OpenDTU-OnBattery nichts in dieser Richtung, die schickt nur einfache Zahlenwerte per MQTT.

**Reply by Tu4r3g21** [reply]
Sorry, da hab ich mich wohl missverständlich ausgedrückt. Die grafische Darstellungen hab ich mit Grafana gemacht. Ich wollte eigentlich nur sagen, ich hatte diese Werte zur Verfügung und konnte sie auswerten. Im HA habe ich diese Werte erst gar nicht und kann auch nichts in ein Diagramm packen.

**Reply by schlimmchen** [reply]
Achso. Ähm, ja, verzeih mir, dass ich grad zu faul bin im Code nachzusehen, aber soweit ich mich erinnere ist es korrekt, dass die Zellspannungen nicht in HA auto-discovery drin sind. Deren Werte werden aber veröffentlicht. Schau mal mit einem Tool wie dem MQTT Explorer auf deinen MQTT Broker, dann wirst du sehen, dass das Werte sind. Wie du an die dann rankommst mit Grafana, kann ich nicht sagen, kommt ja auch auf den Setup an.

**Reply by Tu4r3g21** [reply]
😂👍alles klar. Dachte mir doch dass das Problem bei mir vor dem Monitor sitzt...
Aber danke für den Hinweis, dann weiß ich jetzt das ich einfach nur ein wenig tiefer buddeln muss!

**Comment by clausthaler** [comment]
Servus,

vielen Danke für dieses Projekt und die Arbeit, die ihr da reinsteckt! Habe bei mir gerade die OpenDTU On Battery Erweiterung auf einem Fusion V2 Board installiert und die Daten eines JKPB2AS1620P werden angezeigt.
Unterstützt diese Erweiterung mehrere parallele JK-BMS? Sorry wenn die Frage schon mal irgendwo beantwortet wurde, hab nichts dazu gefunden. Hab nur gesehen, dass es mit parallelen Pylontechs geht.
Habe aktuell zwei Akkus parallel, geplant sind 4. 
Besten Dank und VG

**Reply by schlimmchen** [reply]
> Unterstützt diese Erweiterung mehrere parallele JK-BMS?

Nope. Allerdings weiß ich auch nicht, wie die sich verhalten parallel geschaltet, bzw. was es heißt, dass die parallel geschaltet sind. Was heißt das denn?

**Reply by clausthaler** [reply]
Es sind mehrere Akkus parallel geschaltet, jeweils mit eigenem JK BMS, die untereinander per RS485 kommunizieren. Siehe Manual https://www.jkbms.com/wp-content/uploads/2024/06/JK-BMS-User-Manual-for-PB-series-jkbms.com_.pdf unter "Batteries in parallel"

**Comment by m4gicSmok13y** [comment]
Moin Moin, ich bin jetzt auch im openDTU Fusion Club und versuche gerade mein JK Inverter BMS anzubinden. Ich habe das CAN/ISO shield im Einsatz. Ich hab das wiki gelesen. Hab aber trotzdem keine Ahnung wo ich jetzt genau mit den Pins meines abgeschnitten Netzwerkkabel ran muss. Über ein Bild wäre ich sehr dankbar.

**Reply by schlimmchen** [reply]
Leider hast du ein paar Dinge nicht verstanden. Du schreibst du hast im Wiki gelesen. Steht da noch etwas außer der Link zu Doku? Bitte schau da mal rein: https://opendtu-onbattery.net/hardware/jkbms/models_pb/ Über CAN kann das JK BMS nicht angebunden werden, sondern es wird über RS485 angebunden.

**Reply by m4gicSmok13y** [reply]
Ich verstehe deine Aussage. Anfangs habe ich versucht, über die Emulation des Pylon-Protokolls via CAN zu kommunizieren, leider jedoch ohne Erfolg. Daraufhin bin ich auf diesen Thread gestoßen.
Mir ist bewusst, dass ich vier Kabel verlegen muss, und ich weiß auch, welche Pins des JK-Ausgangs auf die Pins 16, 15, 45 und 46 der DTU geführt werden müssen. Allerdings habe ich bisher kein Mapping gefunden, das diese Pin-Nummern den tatsächlichen Anschlüssen der DTU zuordnet. Genau dieser Hinweis fehlt mir noch – daher auch meine Bitte nach einem Foto.

**Reply by schlimmchen** [reply]
Wieder nicht verstanden... Du brauchst keine vier Drähte, sondern nur drei, ggf. sogar nur zwei. RS485 A, B und optional GND. Am Fusion ist am PoE-Shield ein RS485 Anschluss mit Masse (GND) drauf. Hier ein Foto: https://github.com/markusdd/OpenDTUFusionDocs/blob/main/POE.md

Die ESP32 Pins 16, 15, 45, und 46 gehören ins Mapping, aber die gehen an den Transceiver, der schon auf dem Fusion verlötet ist. So ist es auch in der Doku in der ASCII Art angedeutet. Mit diesen Pins hast du nichts zu tun, außer das sie ins pin_mapping.json müssen.

**Reply by thankthemaker** [reply]
Is there a way to get RS-485 working with JK-BMS without the PoE-Shield? I also do have the CAN/ISO-Shield, not using it for now.

**Reply by AndreasBoehm** [reply]
You don't need the PoE-Shield, the RS-485 transceiver is available via two pins on the fusion board (A and B) on the left side of the board, right next to the BOOT button: https://github.com/markusdd/OpenDTUFusionDocs/tree/main#opendtu-fusion-documentation

**Comment by thankthemaker** [comment]
@AndreasBoehm Thank you very much. Great news!

**Comment by bikerbiker12** [comment]
Hallo zusammen,

ich klinke mich hier mal ein. Habe das FusionBoard schon länger in Betrieb und musste jetzt auf ein JK BMS PB umstellen. Anbindung über den RS485 vom JK Kommunication Board PIN 1+2 auf den Anschluss am Fusion. Protokoll in der OpenDTU wie nach Anleitung eingestellt. Läuft alles Prima! Allerdings fehlen mir ein paar Werte in der MQTT oder auch in der Auto-Discovery. Der wichtigste wäre aktuell "Leistung in W" von der Batterie
in der OpenDtu Live Ansicht wird dieser Wert ja ausgegeben, ist dieser kalkuliert? Muss im BMS etwas um- oder eingestellt werden, dass dieser Wert übermittelt wird? Als Protokoll im BMS habe ich das "GPS.." ausgewählt, alle anderen liefern gar keine Werte?

Liegt da ein Fehler meinerseits vor, oder wird der Wert einfach nicht übergeben?

Grüße Marco
